src/Build/Construction/ProjectRootElement.cs

```diff
@@ -176,6 +176,14 @@ internal ProjectRootElement(XmlReader xmlReader, ProjectRootElementCacheBase pro
             ProjectParser.Parse(document, this);
         }
 
+        private readonly bool _cannotBeDirtied = false;
+
+        private ProjectRootElement(ProjectRootElementCacheBase projectRootElementCache, NewProjectFileOptions projectFileOptions, bool canBeDirty)
+            : this(projectRootElementCache, projectFileOptions)
+        {
+            _cannotBeDirtied = canBeDirty;
```

Comment:
Consider naming variables in way that implies "presence" rather than "absence", as it's much more mentally taxing to trying to imagine an absence of something vs having something. There have been few researched done in this area.

Another issue here is readability and comprehension - we're assigning here a variable which "can be dirty" to something that "can't be dirty". It took me several minutes to try to process this, and yet I'm still unsure what this means...

Reply:
thanks! I name it `canBeDirty` then later changed to `cannotbeDirtied` and missed fixing this part.
I'll rename everything back to `canBeDirty`